src/prototype_room_costmatrix.js

```diff
@@ -57,41 +57,47 @@ Room.prototype.setCostMatrixPath = function(costMatrix, path) {
     let pos = path[i];
     costMatrix.set(pos.x, pos.y, config.layout.pathAvoid);
   }
+  this.setCostMatrixAvoidSources(costMatrix);
```

Comment:
Yes, true, but having a low avoid value on the path is fine. That the paths should avoid sources is handled when creating the path. When using `moveToMy` the creeps 'should' walk on the paths.
So if a creep wants to move to the source if should move on the path, if it moves somwhere else due to the surrounding high avoid values, it shouldn't move on that path anyway.
Makes sense?